.github/workflows/conformance-ingress-shared.yaml

```diff
@@ -0,0 +1,165 @@
+    timeout-minutes: 120
+    steps:
+      - name: Checkout master branch to access local actions
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+        with:
+          ref: ${{ github.event.repository.default_branch }}
+          persist-credentials: false
+      - name: Set Environment Variables
+        uses: ./.github/actions/set-env-variables
+
+      - name: Install Cilium CLI
+        run: |
+          curl -sSL --remote-name-all https://github.com/cilium/cilium-cli/releases/download/${{ env.cilium_cli_version }}/cilium-linux-amd64.tar.gz{,.sha256sum}
+          sha256sum --check cilium-linux-amd64.tar.gz.sha256sum
+          sudo tar xzvfC cilium-linux-amd64.tar.gz /usr/local/bin
+          rm cilium-linux-amd64.tar.gz{,.sha256sum}
+          cilium version
+
+      - name: Set image tag
+        id: vars
+        run: |
+          if [ ${{ github.event.pull_request }} ]; then
+            SHA=${{ github.event.pull_request.head.sha }}
+          else
+            SHA=${{ github.sha }}
+          fi
+          echo ::set-output name=sha::${SHA}
+
+      - name: Checkout
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+        with:
+          ref: ${{ steps.vars.outputs.sha }}
+          persist-credentials: false
+
+      - name: Create minikube cluster with multiple nodes
+        run: |
+          # Use at least 3 nodes, so that at least 2 worker nodes are available for scheduling
+          minikube start --network-plugin=cni --cni=false \
+            --docker-opt="default-ulimit=nofile=102400:102400" \
+            --nodes 3
+
+      # Start minikube tunnel
+      - name: Setup minikube
+        run: |
+          docker pull ${{ env.ECHO_SERVER_IMAGE }}
+          minikube image load ${{ env.ECHO_SERVER_IMAGE }}
+          minikube tunnel &
+        env:
+          ECHO_SERVER_IMAGE: k8s.gcr.io/ingressconformance/echoserver:v0.0.1@sha256:9b34b17f391f87fb2155f01da2f2f90b7a4a5c1110ed84cb5379faa4f570dc52
+
+      - name: Checkout ingress-controller-conformance
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+        with:
+          repository: kubernetes-sigs/ingress-controller-conformance
+          path: ingress-controller-conformance
+          ref: a38bfcddfc7b10a7b479c1f274b87380c7390c96
+          persist-credentials: false
+
+      - name: Install Ingress conformance test tool
+        timeout-minutes: 10
+        run: |
+          cd ingress-controller-conformance
+          make build
+
+      - name: Wait for images to be available
+        timeout-minutes: 30
+        shell: bash
+        run: |
+          for image in cilium-ci operator-generic-ci ; do
+            until docker manifest inspect quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/$image:${{ steps.vars.outputs.sha }} &> /dev/null; do sleep 45s; done
+          done
+
+      - name: Install Cilium
+        run: |
+          helm upgrade -i cilium ./install/kubernetes/cilium \
```

Comment:
> Why upgrade instead of install?

Probably it's just my habit to use `upgrade -i` locally as it will work for both fresh installation and upgrade (which is useful for local testing)

> Also, why not use cilium-cli? Is this a holdover from the conformance-ingress.yaml
Good point, this job is the same as conformance-ingress (with one more helm param changes ingressController.loadbalancerMode=shared). I think when I added this conformance test, the cilium-cli didn't support helm chart 🤔